code/modules/admin/verbs/randomverbs.dm

```diff
@@ -349,111 +349,117 @@ Traitors and the like can also be revived with the previous role mostly intact.
 /N */
 /client/proc/respawn_character()
 	set category = "Special Verbs"
-	set name = "Respawn Character"
-	set desc = "Respawn a person that has been gibbed/dusted/killed. They must be a ghost for this to work and preferably should not have a body to go back into."
+	set name = "Spawn Character"
+	set desc = "(Re)Spawn a client's loaded character."
 	if(!holder)
 		src << "Only administrators may use this command."
 		return
-	var/input = ckey(input(src, "Please specify which key will be respawned.", "Key", ""))
-	if(!input)
-		return
 
-	var/mob/observer/dead/G_found
-	for(var/mob/observer/dead/G in player_list)
-		if(G.ckey == input)
-			G_found = G
-			break
+	//I frontload all the questions so we don't have a half-done process while you're reading.
+	var/client/picked_client = input(src, "Please specify which client's character to spawn.", "Client", "") in clients
+	if(!picked_client)
+		return
 
-	if(!G_found)//If a ghost was not found.
-		usr << "<font color='red'>There is no active key like that in the game or the person is not currently a ghost.</font>"
+	var/location = input(src,"Please specify where to spawn them.", "Location", "Right Here") in list("Right Here","Arrivals","Cancel")
+	if(!location || location == "Cancel")

```

Comment:
Hm, I added an option to spawn you at job spawn locations, but after the round starts and everyone spawns, it seems that it deletes all the spawn points. So, going to say out of scope on that one. Just aghost where you want them and spawn them.